evm_loader/program/src/evm/tracing/event_listener/tracer.rs

```diff
@@ -1,13 +1,19 @@
+use std::sync::{Arc, RwLock};
+
+use crate::evm::tracing::event_listener::trace::{FullTraceData, VMTrace, VMTracer};
+
 use super::vm_tracer::VmTracer;
-use crate::types::trace::{FullTraceData, VMTrace, VMTracer};
+
+pub type TracerType = Option<Arc<RwLock<Option<Tracer>>>>;
```

Comment:
Do you really need TracerType to be so complicated? Where do you use it from the multiple threads? Also, what's the reason for the second option?

Reply:
Please check the end of the PR description for why this type is so complicated

Reply:
Yes, `emulate_trx` is single-threaded because it uses `block()` calls deep inside its implementation when using `EmulatorAccountStorage`. The `block()` calls are needed because there is a non-async interface (`AccountStorage` trait) between two async contexts (`axum` and Solana or ClickHouse client). But ideally even the opcode functions should be async in this implementation.

This was my initial solution too here: https://github.com/neonlabsorg/neon-evm/pull/144.

Why I chose this solution in the end is because we need to remove those `block()` calls from the EVM used inside `neon-api` because they affect performance. This is a step in that direction too, unless we decide to use `Actix` instead of `axum`.

We should decide which of the two PRs we should merge.

Reply:
Ok, I support future implementation of `maybe_async`